### PR TITLE
chore(ci/rust): build and test more packages in Windows

### PR DESCRIPTION
**Pull request by ReactorScram**

### CODE REVIEW

**Comment by ReactorScram** [comment]
@thomaseizinger I assume we weren't leaving these out for any reason?

File: rust/tun/Cargo.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ version = "0.1.0"
 edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
-[dependencies]
+[target.'cfg(target_os = "windows")'.dependencies]
```
Comment (ReactorScram) [inline rust/tun/Cargo.toml]: The unused deps step in CI picked this up

**Review by thomaseizinger — APPROVED** [review]
I think the intention originally was to only include packages that are either:

- leaf crates, i.e. don't get compiled as a transitive dependency otherwise
- have meaningful, platform-specific tests

I guess it can't hurt to test more of them on more platforms.

Ideally, we just test the entire workspace on all of them and either make crates actually work on all platforms or add stubs where necessary.

File: rust/tun/Cargo.toml [hunk 2]
```diff
@@ -4,7 +4,7 @@ version = "0.1.0"
 edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
-[dependencies]
+[target.'cfg(target_family = "unix")'.dependencies]
```
Comment (ReactorScram) [inline rust/tun/Cargo.toml:7]: Satisfy the unused dep checker

**Pull request merged**